scripts/configure-dh.sh

```diff
@@ -0,0 +1,363 @@
+#!/bin/bash
+
+# Constants
+BACKSTAGE_CR_NAME="ai-rh-developer-hub"
+DEFAULT_RHDH_DEPLOYMENT="backstage-ai-rh-developer-hub" # deployment created by rhdh operator by default
+PLUGIN_CONFIGMAP="dynamic-plugins" # configmap created by rhdh operator for plugins by default
+EXTRA_ENV_SECRET="ai-rh-developer-hub-env" # secret created by rhdh installer to store private env vars
+
+# Variables
+BASE_DIR="$(realpath $(dirname ${BASH_SOURCE[0]}))/.."
+CATALOGS_FILE=${CATALOGS_FILE:-"${BASE_DIR}/catalogs.yaml"}
+RHDH_DEPLOYMENT="${DEFAULT_RHDH_DEPLOYMENT}"
+RHDH_EXTRA_ENV_SECRET="${EXTRA_ENV_SECRET}"
+RHDH_PLUGINS_CONFIGMAP="${PLUGIN_CONFIGMAP}"
+NAMESPACE=${NAMESPACE:-"ai-rhdh"}
+EXISTING_NAMESPACE=${EXISTING_NAMESPACE:-''}
+EXISTING_DEPLOYMENT=${EXISTING_DEPLOYMENT:-''}
+EXISTING_EXTRA_ENV_SECRET=${EXISTING_EXTRA_ENV_SECRET:-''}
+RHDH_PLUGINS=${RHDH_PLUGINS:-''}
+RHDH_INSTANCE_PROVIDED=${RHDH_INSTANCE_PROVIDED:-false}
+RHDH_GITHUB_INTEGRATION=${RHDH_GITHUB_INTEGRATION:-false}
+RHDH_GITLAB_INTEGRATION=${RHDH_GITLAB_INTEGRATION:-false}
+
+# Secret variables
+GITHUB__APP__ID=${GITHUB__APP__ID:-''}
+GITHUB__APP__CLIENT__ID=${GITHUB__APP__CLIENT__ID:-''}
+GITHUB__APP__CLIENT__SECRET=${GITHUB__APP__CLIENT__SECRET:-''}
+GITHUB__APP__WEBHOOK__URL=${GITHUB__APP__WEBHOOK__URL:-''}
+GITHUB__APP__WEBHOOK__SECRET=${GITHUB__APP__WEBHOOK__SECRET:-''}
+GITHUB__APP__PRIVATE_KEY=${GITHUB__APP__PRIVATE_KEY:-''}
+GITOPS__GIT_TOKEN=${GITOPS__GIT_TOKEN:-''}
+GITLAB__APP__CLIENT__ID=${GITLAB__APP__CLIENT__ID:-''}
+GITLAB__APP__CLIENT__SECRET=${GITLAB__APP__CLIENT__SECRET:-''}
+GITLAB__TOKEN=${GITLAB__TOKEN:-''}
+QUAY__DOCKERCONFIGJSON=${QUAY__DOCKERCONFIGJSON:-''}
+QUAY__API_TOKEN=${QUAY__API_TOKEN:-''}
+
+# Use existing variables if RHDH instance is provided
+if [[ $RHDH_INSTANCE_PROVIDED != "true" ]] && [[ $RHDH_INSTANCE_PROVIDED != "false" ]]; then
+    echo -n "RHDH_INSTANCE_PROVIDED needs to be set to either 'true' or 'false'"
+    echo "FAIL"
+    exit 1
+elif [[ $RHDH_INSTANCE_PROVIDED == "true" ]]; then
+    NAMESPACE="${EXISTING_NAMESPACE}"
+    RHDH_DEPLOYMENT="${EXISTING_DEPLOYMENT}"
+    RHDH_EXTRA_ENV_SECRET="${EXISTING_EXTRA_ENV_SECRET}"
+    RHDH_PLUGINS_CONFIGMAP="${RHDH_PLUGINS}"
+fi
+
+# Reading secrets
+# Reads secrets either from environment variables or user input
+echo -n "* Reading secrets: "
+
+# Reads GitHub secrets if enabling GitHub integration
+if [[ $RHDH_GITHUB_INTEGRATION == "true" ]]; then
+    # Reads GitHub Org App ID
+    until [ ! -z "${GITHUB__APP__ID}" ]; do
+        read -p "Enter your GitHub App ID: " GITHUB__APP__ID
```

Comment:
Some of these info is being asked a second time, the first time being in configure_pipelines.sh

Can we streamline this? I feel a main script like configure.sh should take all this info rather than asking twice in individual scripts..

Reply:
@maysunfaisal resolved with https://github.com/redhat-ai-dev/ai-rhdh-installer/pull/10/commits/83d0eba4b9e076fe25ea1ef8f2e127229db9353d

User is now prompted once under the `configure.sh` script instead of prompted for each script the input is used.